src/providers/theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import {
+  CssBaseline,
+  ThemeProvider as MUIThemeProvider,
+  PaletteMode,
+  createTheme,
+} from '@mui/material';
+import { ReactNode, useMemo, useState } from 'react';
+import { ThemeContext, ThemeContextProps } from './ThemeContext';
+import { getTheme } from './theme';
+
+type ThemeProviderProps = {
+  children: ReactNode;
+};
+
+export const ThemeProvider = ({ children }: ThemeProviderProps): JSX.Element => {
+  const [mode, setMode] = useState<PaletteMode>('light');
```

Comment:
Please use user preferred mode as initial value for theme. Like this: `const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');`